contracts/Governance/Timelock.sol

```diff
@@ -1,123 +1,160 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.13;
+pragma experimental ABIEncoderV2;
 
-contract Timelock {
-    event NewAdmin(address indexed newAdmin);
-    event NewPendingAdmin(address indexed newPendingAdmin);
-    event NewDelay(uint256 indexed newDelay);
-    event CancelTransaction(bytes32 indexed txHash, address indexed target, uint256 value, string signature, bytes data, uint256 eta);
-    event ExecuteTransaction(bytes32 indexed txHash, address indexed target, uint256 value, string signature, bytes data, uint256 eta);
-    event QueueTransaction(bytes32 indexed txHash, address indexed target, uint256 value, string signature, bytes data, uint256 eta);
+import "@openzeppelin/contracts/access/Ownable.sol";
 
-    uint256 public constant GRACE_PERIOD = 14 days;
-    uint256 public constant MINIMUM_DELAY = 2 days;
-    uint256 public constant MAXIMUM_DELAY = 30 days;
+contract Timelock is Ownable {
+    enum TrnasactionStatus {
+        NonExistent,
+        Queued,
+        Approved
+    }
+
+    struct Transaction {
+        address target;
+        uint256 value;
+        string signature;
+        bytes data;
+    }
+
+    uint256 public minimumDelay;
+    uint256 public maximumDelay;
+    uint256 public gracePeriod;
+
+    mapping(bytes32 => TrnasactionStatus) public queuedTransactions;
 
-    address public admin;
-    address public pendingAdmin;
+    address public proposer;
     uint256 public delay;
 
-    mapping(bytes32 => bool) public queuedTransactions;
+    event QueuedTransactionsBatch(bytes32 indexed txParamsHash, uint256 numberOfTransactions, uint256 eta);
+    event CancelledTransactionsBatch(bytes32 indexed txParamsHash);
+    event ApprovedTransactionsBatch(bytes32 indexed txParamsHash);
+    event ExecutedTransaction(bytes32 indexed txParamsHash, address indexed target, uint256 value, string signature, bytes data, uint256 eta);
+    event NewProposerSet(address indexed newProposer);
+    event NewDelaySet(uint256 indexed delay);
+
+    constructor(
+        address _proposer,
+        uint256 _minimumDelay,
+        uint256 _maximumDelay,
+        uint256 _gracePeriod,
+        uint256 _delay
+    ) {
+        require(_proposer != address(0), "Proposer address cannot be 0");
+        require(_minimumDelay <= _maximumDelay, "Minimum delay cannot be larger than maximum delay");
+
+        minimumDelay = _minimumDelay;
+        maximumDelay = _maximumDelay;
```

Comment:
Let's add a setter for the max and min delays. The min setter should not accept values smaller than 1 day.